### PR TITLE
job-dsl: framework to support in fleet-ci

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use job-dsl to create jobs without the need for JJBB

## Why is it important?

Use same repository in different places

## Next

Requires the JJB in the infra side

### CODE REVIEW

File: .ci/jobdsl-generator.groovy [hunk 1]
```diff
@@ -78,7 +78,7 @@ pipeline {
           removedJobAction: 'DELETE',
           removedViewAction: 'DELETE',
           sandbox: true,
-          targets: "${BASE_DIR}/.ci/jobDSL/jobs/**/folder.groovy",
+          targets: "${BASE_DIR}/.ci/jobDSL/jobs/apm-ci/folder.groovy",
```
Comment (v1v) [inline .ci/jobdsl-generator.groovy:81]: This should avoid issues with pointing to a wrong CI controller

File: .ci/jobdsl-generator.groovy [hunk 2]
```diff
@@ -89,7 +89,7 @@ pipeline {
           removedJobAction: 'DELETE',
           removedViewAction: 'DELETE',
           sandbox: true,
-          targets: "${BASE_DIR}/.ci/jobDSL/jobs/**/*.groovy",
+          targets: "${BASE_DIR}/.ci/jobDSL/jobs/apm-ci/**/*.groovy",
```
Comment (v1v) [inline .ci/jobdsl-generator.groovy:92]: This should avoid issues with pointing to a wrong CI controller

File: .ci/jobdsl-fleet-ci.groovy [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      steps {
+        jobDsl(
+          ignoreExisting: true,
+          failOnMissingPlugin: true,
+          failOnSeedCollision: true,
+          removedConfigFilesAction: 'DELETE',
+          removedJobAction: 'DELETE',
+          removedViewAction: 'DELETE',
+          sandbox: true,
+          targets: "${BASE_DIR}/.ci/jobDSL/jobs/fleet-ci/**/folder.groovy",
```
Comment (v1v) [inline .ci/jobdsl-fleet-ci.groovy:81]: This should avoid issues with pointing to a wrong CI controller

File: .ci/jobdsl-fleet-ci.groovy [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        )
+        jobDsl(
+          ignoreExisting: true,
+          failOnMissingPlugin: true,
+          failOnSeedCollision: true,
+          removedConfigFilesAction: 'DELETE',
+          removedJobAction: 'DELETE',
+          removedViewAction: 'DELETE',
+          sandbox: true,
+          targets: "${BASE_DIR}/.ci/jobDSL/jobs/fleet-ci/**/*.groovy",
```
Comment (v1v) [inline .ci/jobdsl-fleet-ci.groovy:92]: This should avoid issues with pointing to a wrong CI controller

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**